### DISCUSSION TITLE
Minimized taskbar apps not being auto restored

### DISCUSSION THREAD

**Discussion by 80sreplicant** [kangyu-california/PersistentWindows#239] [Q&A] [answered]
So I've been having some windows still being sent to my second monitor after my monitors wake up from being automatically turned off. What I noticed is that it's programs that have the option to minimize to system tray when you close out on them, for example steam and qbittorrent. I usually close those programs but have them running in the taskbar, I close them on my main monitor but when my monitors turn on they automatically are on my second monitor. Is there any fix for these minimized programs to get sent to the right monitor?

Just in case, both monitors are using Display ports and I'm using windows 11 22h2 with the latest update.

**Comment by kangyu-california** [comment]
First please try the latest 5.42 release which resolved #224 (minimized window icon not placed to correct taskbar)

Some applications such as winamp, does not use standard window style, therefore when you "close out" them, PW would not get notified of the minimize event, hence the restore would be inaccurate.
You may find out what happened by running PW with -debug_process "steam" command option to print out all events of specified process in event viewer.

**Comment by 80sreplicant** [comment]
This is the event log I get with qbittorrent. Also, what I meant with minimizing the app to taskbar, I meant when I close these two applications, they both are still running in my system tray.

![mmc_TaRdKnULta](https://github.com/kangyu-california/PersistentWindows/assets/47994283/4a8df3c8-6536-4eef-923f-2e09af46f2b9)

**Comment by kangyu-california** [comment]
PW remembers the minimized state if you give it enough time (4~5 seconds) to capture before switching monitor off
![image](https://github.com/kangyu-california/PersistentWindows/assets/59128756/64683f09-7140-4b1d-8c84-643c588985c8)

However either Windows OS or the app itself may revert window position that PersistentWindows has restored due to lack of coordination. The conflict can be resolved by adding delay to restore via command line argument such as "-delay_auto_restore 5" when launching PersistentWindows.exe

**Comment by 80sreplicant** [comment]
That doesn't seem to have worked, ill add a longer delay and test it again.

Edit: Never mind, they still open on second monitor and even with the delay argument won't switch to main monitor.

**Reply by kangyu-california** [reply]
the issue is tracked in #240

**Reply by kangyu-california** [reply]
Please try [the latest patch](https://github.com/kangyu-california/PersistentWindows/files/12065409/PersistentWindows5.42_p240.zip)

**Reply by 80sreplicant** [reply]
I just tried the patch, nothing changed, both applications still open on second monitor when waking up from being off.

![mmc_4mtAyQRA88](https://github.com/kangyu-california/PersistentWindows/assets/47994283/2dc7eef0-5007-4647-a69b-a775a6a2b2ef)

**Reply by kangyu-california** [reply]
Not sure if you are running PW with admin privilege or not. Run with admin mode is recommended.

**Reply by 80sreplicant** [reply]
I do have it running in admin mode,

**Comment by 80sreplicant** [comment]
Heres a video of my event viewer, just in case you need more detail

https://github.com/kangyu-california/PersistentWindows/assets/47994283/40a7aaa6-e4d8-44e5-afc0-8782a7713151

**Reply by kangyu-california** [reply]
The event log really helps. Please try [PersistentWindows5.43_p239.zip](https://github.com/kangyu-california/PersistentWindows/files/12074560/PersistentWindows5.43_p239.zip)

**Reply by 80sreplicant** [reply]
Didn't work, here is my event log


https://github.com/kangyu-california/PersistentWindows/assets/47994283/c6071f99-2d37-4d2e-9a64-064e6f911382

Detailed



https://github.com/kangyu-california/PersistentWindows/assets/47994283/b1522d9a-6f07-42c4-b85c-2a0192189036

Just wanted to put in some more clarity. Both of my monitors are using display port, my second monitor wakes up faster than my main monitor, and I'm assuming because of this PA cant bring both steam and qbittorrent to the main monitor fast enough, other windows do get pushed to the main monitor, but I'm assuming because these two get pushed to the system tray when minimized, that effects it.

**Reply by kangyu-california** [reply, marked as answer]
The latest patch [PersistentWindows5.43_p239_240_241.zip](https://github.com/kangyu-california/PersistentWindows/files/12139138/PersistentWindows5.43_p239_240_241.zip)

**Reply by 80sreplicant** [reply]
I have tested it twice with this patch and it seems to work!
